docs/docs/troubleshooting.md

```diff
@@ -0,0 +1,165 @@
+
+1. Download the `jwks.json` file from the authenticate server:
+
+    ::::: tabs
+    :::: tab curl
+
+    ```bash
+    curl https://authenticate.localhost.pomerium.io/.well-known/pomerium/jwks.json > /etc/grafana/jwks.json
+    ```
+    ::::
+    :::: tab wget
+
+    ```bash
+    wget -O /etc/grafana/jwks.json https://authenticate.localhost.pomerium.io/.well-known/pomerium/jwks.json
+    ```
+    ::::
+    :::::
+
+1. Edit the upstream service configuration to use the local key to verify tokens:
+
+    ```ini
+    [auth.jwt]
+    enabled = true
+    header_name = X-Pomerium-Jwt-Assertion
+    email_claim = email
+    jwk_set_file = /etc/grafana/jwks.json
+    cache_ttl = 60m
+    ```
+
+### Kubernetes Ingress Controller
+
+#### View Event History
+
+Pomerium Ingress Controller will add **events** to the Ingress objects as it processes them.
+
+```
+kubectl describe ingress/my-ingress
+```
+
+```log
+Events:
+  Type    Reason   Age   From              Message
+  ----    ------   ----  ----              -------
+  Normal  Updated  18s   pomerium-ingress  updated pomerium configuration
+```
+
+If an error occurs, it may be reflected in the events:
+
+```log
+Events:
+  Type     Reason       Age                 From              Message
+  ----     ------       ----                ----              -------
+  Normal   Updated      5m53s               pomerium-ingress  updated pomerium configuration
+  Warning  UpdateError  3s                  pomerium-ingress  upsert routes: parsing ingress: annotations: applying policy annotations: parsing policy: invalid rules in policy: unsupported conditional "maybe", only and, or, not, nor and action are allowed
+```
+
+### Redirect Loop with Redis Databroker
+
+When using Redis, the [shared secret](/reference/readme.md#shared-secret) is used to encrypt data in Redis itself. if you change the configured shared secret, data from Redis can no longer be decrypted. This results in errant behavior, including redirect loops when a user session cannot be retrieved from the databroker.
+
+The resolution is to flush the Redis database with [`FLUSHDB`](https://redis.io/commands/flushdb) or [`FLUSHALL`](https://redis.io/commands/FLUSHALL).
+
+### RPC Errors
+
+#### certificate signed by unknown authority
+
+When authenticating and authorizing a user for the first time, you may see the following in your Pomerium logs:
+
+```log
+ERR http-error error="401 Unauthorized: ..... rpc error: code = DeadlineExceeded desc = latest connection error: connection error: desc = "transport: authentication handshake failed: x509: certificate signed by unknown authority...."
+```
+
+**Why**
+
+This error means that the proxy is rejecting the authorize service's supplied certificate (used to establish a secure connection) because it doesn't know or trust the certificate authority that is associated with the supplied certificate. This can happen for a few reasons.
```

Comment:
```suggestion
This error means that the proxy is rejecting the authorized service's supplied certificate (used to establish a secure connection) because it doesn't know or trust the certificate authority that signed it. This can happen for a few reasons.
```
Regarding "This can happen for a few reasons" I propose explaining at least some of those reasons or just cutting it out.

Reply:
:+1:  on reducing to "signed it", but the "authorize service" is correct. Pomerium is broken down into components: authorize, proxy, and the data broker. So in this case this isn't an adjective but a noun. It would be more obvious if it were a proper noun "Authorize service".... CC @desimone to validate that idea.

```suggestion
This error means that the proxy is rejecting the Authorize service's supplied certificate (used to establish a secure connection) because it doesn't know or trust the certificate authority that signed it. This can happen for a few reasons.
```

> I propose explaining at least some of those reasons

I feel like this concern is addressed by the **Solution** section, which covers possible resolutions for several possible reasons. But if we don't have consensus on that idea, then cutting it out works for me.

```suggestion
This error means that the proxy is rejecting the authorize service's supplied certificate (used to establish a secure connection) because it doesn't know or trust the certificate authority that signed it.
```